test/integrations/sources/webhook/data.ts

```diff
@@ -0,0 +1,227 @@
+                writeKey: ['write_key'],
+              },
+            },
+            source: {
+              Config: {
+                putRequestDetailsInContext: true,
+              },
+            },
+          },
+          {
+            request: {
+              body: JSON.stringify({
+                prop1: 'value1',
+                prop2: 'value2',
+              }),
+              method: 'POST',
+              url: '/v1/webhook?writeKey=write_key&query_param_key_1=query_param_value_1&query_param_key_2=query_param_value_2',
+              proto: 'HTTP/1.1',
+              headers: {
+                Connection: ['keep-alive'],
+                'Content-Length': ['48'],
+                'Content-Type': ['application/json'],
+              },
+              query_parameters: {
+                query_param_key_1: ['query_param_value_1'],
+                query_param_key_2: ['query_param_value_2'],
+                writeKey: ['write_key'],
+              },
+            },
+            source: {
+              Config: {
+                putRequestDetailsInContext: false,
+              },
+            },
+          },
+          {
+            request: {
+              body: JSON.stringify({
+                prop1: 'value1',
+                prop2: 'value2',
+              }),
+              method: 'POST',
+              url: '/v1/webhook?writeKey=write_key&query_param_key_1=query_param_value_1&query_param_key_2=query_param_value_2',
+              proto: 'HTTP/1.1',
+              headers: {
+                Connection: ['keep-alive'],
+                'Content-Length': ['48'],
+                'Content-Type': ['application/json'],
+              },
+              query_parameters: {
+                query_param_key_1: ['query_param_value_1'],
+                query_param_key_2: ['query_param_value_2'],
+                writeKey: ['write_key'],
+              },
+            },
+            source: {},
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            output: {
+              batch: [
+                {
+                  type: 'track',
+                  event: 'webhook_source_event',
+                  properties: {
+                    prop1: 'value1',
+                    prop2: 'value2',
+                  },
+                  anonymousId: ANONYMOUS_ID,
+                  context: {
+                    method: 'POST',
+                    url: '/v1/webhook?writeKey=write_key&query_param_key_1=query_param_value_1&query_param_key_2=query_param_value_2',
+                    proto: 'HTTP/1.1',
+                    headers: {
+                      Connection: ['keep-alive'],
+                      'Content-Length': ['48'],
+                      'Content-Type': ['application/json'],
+                    },
+                    query_parameters: {
+                      query_param_key_1: ['query_param_value_1'],
```

Comment:
product requirement was to send them as simple data types - both query params & headers